### PR TITLE
Support node upgrade procedure

### PR DESCRIPTION
**Pull request by juliayakovlev**
Added support of node upgrade and method that upgrade
whole cluster

I opened this PR as Draft because of I want to run byo dtest and validate that it doesn't broke other tests

### CODE REVIEW

File: ccmlib/scylla_node.py [hunk 1]
```diff
@@ -1141,3 +1178,97 @@ def no_pending_flushes():
...
+
+    def upgrade(self, upgrade_to_version):
+        self.upgrader.upgrade(upgrade_version=upgrade_to_version)
+
+class NodeUpgrader:
+
+    """
+    Upgrade node is supported when uses relocatable packages only
+    """
+    copy_suffix = '.org'
```
Comment (fruch) [inline ccmlib/scylla_node.py:1205]: where this is used ? and for what ?

File: ccmlib/scylla_node.py [hunk 2]
```diff
@@ -795,52 +821,62 @@ def import_dse_config_files(self):
...
+            os.remove(dst)
+
         try:
             os.link(src, dst)
         except OSError as oserror:
             if oserror.errno == errno.EXDEV or oserror.errno == errno.EMLINK:
                 do_copy(src, dst, extra_perms)
             else:
-                raise RuntimeError("Unable to create hard link from %s to %s: %s" % (src, dst, oserror))
+                 raise RuntimeError("Unable to create hard link from %s to %s: %s" % (src, dst, oserror))
```
Comment (fruch) [inline ccmlib/scylla_node.py:841]: why the extra space here ?

**Review by fruch — APPROVED** [review]
LGTM

File: ccmlib/scylla_node.py (hunk 1 above)
Comment (juliayakovlev) [reply ccmlib/scylla_node.py]: removed

File: ccmlib/scylla_node.py (hunk 2 above)
Comment (juliayakovlev) [reply ccmlib/scylla_node.py]: fixed

**Comment by fruch** [comment]
@juliayakovlev you got it working in BYO  ? (would be nice if you have a link, just for the confirmation)

File: ccmlib/scylla_node.py [hunk 3]
```diff
@@ -1141,3 +1178,95 @@ def no_pending_flushes():
...
+        except Exception as exc:
+            self.node.node_install_dir = self.orig_install_dir
+            raise NodeUpgradeError("Failed to import executables files. %s" % exc)
+
+    def upgrade(self, upgrade_version: str):
+        """
+        :param upgrade_version: relocatables folder. Example: unstable/master:2020-11-18T08:57:53Z
+        """
+        self.scylla_version_for_upgrade = upgrade_version
+        cdir = self._setup_relocatable_packages()
```
Comment (amoskong) [inline ccmlib/scylla_node.py:1246]: do you plan to add a rollback()?

Comment (juliayakovlev) [reply ccmlib/scylla_node.py:1246]: yes, I have such task

**Comment by juliayakovlev** [comment]
> @juliayakovlev you got it working in BYO ? (would be nice if you have a link, just for the confirmation)

Yes, I ran byo to validat, that it doesn't broke dtest
Byo test: https://jenkins.scylladb.com/view/master/job/scylla-master/job/byo/job/dtest-byo/363/

One test failed but it's not because of this CCM change

**Comment by juliayakovlev** [comment]
@fruch  Can we merge it?

**Pull request merged**